### ISSUE TITLE
Use multicall to batch constant web3 calls in surrogeth-client

### ISSUE THREAD

**Issue by lsankar4033** [lsankar4033/surrogeth#15]
https://github.com/makerdao/multicall

**Comment by barryWhiteHat** [comment]
I belive this will make the transaction simulation more difficult. Imagine i am monirotring the tx pool to see if you have repriced any trnasactions. Without this change I would only need to simulate every transaction that calls 0xrelayer but after this change i would have no way of knowing which transactions eventually call the relayRegistry and i wont know which transactions to simulate unless i do some more advanced analysis

**Comment by weijiekoh** [comment]
Ah, this isn't about transaction simulation, but for clients (like UIs) to aggregate calls to the registry contract. e.g. instead of making one call per  `relayerId` to get a list of `relayerAddress`, this library combines all these calls into one which returns an array.

**Comment by lsankar4033** [comment]
yep, as @weijiekoh mentioned, this is just for reads (not tx-creating methods)

**Comment by barryWhiteHat** [comment]
:+1:
